x-pack/plugins/ml/common/util/parse_interval.ts

```diff
@@ -60,3 +61,18 @@ export function parseInterval(interval: string, checkValidEsUnit = false): Durat
     return null;
   }
 }
+
+/**
+ * Resolves the longest bucket span from the list and multiply it by 2.
+ * @param bucketSpans Collection of bucket spans
+ */
+export function resolveBucketSpanInSeconds(bucketSpans: string[]): number {
```

Comment:
This function should be moved to `common/util/job_utils.ts` as it is very specific to anomaly detection jobs. That file already contains a related function `parseTimeIntervalForJob`, so seems a logical home for this new function.

Reply:
Refactored in https://github.com/elastic/kibana/pull/92377/commits/e57ef21cc643588307bf271741821af765362ea3